xla/hlo/ir/hlo_instructions.cc

```diff
@@ -1118,6 +1118,26 @@ bool HloAllToAllInstruction::IdenticalSlowPathIgnoringChannelIdValues(
          split_dimension_ == casted_other.split_dimension();
 }
 
+HloCollectiveBroadcastInstruction::HloCollectiveBroadcastInstruction(
+    HloOpcode opcode, const Shape& shape,
+    absl::Span<HloInstruction* const> operands,
+    absl::Span<const ReplicaGroup> replica_groups,
+    const std::optional<int64_t>& channel_id)
+    : HloCollectiveInstruction(opcode, shape, operands, replica_groups,
+                               constrain_layout(), channel_id) {}
```

Comment:
This is the cause of ubsan failure. constrain_layout() is uniitialized. May be this just needs to be false.